### PR TITLE
Smallpatches

### PR DESCRIPTION
**Pull request by InfusOnWoW**

### CODE REVIEW

File: WeakAuras/WeakAuras.lua [hunk 1]
```diff
@@ -78,6 +78,48 @@ function WeakAuras.InternalVersion()
...
+    local prefix = ""
+    local data
+    if currentErrorHandlerId then
+      data = WeakAuras.GetData(currentErrorHandlerId)
+    elseif currentErrorHandlerUid then
+      data = Private.GetDataByUID(currentErrorHandlerUid)
+    end
+    if data then
+      Private.AuraWarnings.UpdateWarning(data.uid, "LuaError", "error",
+        L["This aura has caused a lua error."] .. "\n" .. L["Install the addons bugsack and buggraber for detailed error logs."])
```
Comment (Stanzilla) [inline WeakAuras/WeakAuras.lua:95]: L["This aura has caused a Lua error."] .. "\n" .. L["Install the addons BugSack and BugGrabber for detailed error logs."])

File: WeakAuras/GenericTrigger.lua [hunk 2]
```diff
@@ -4053,28 +4052,28 @@ function GenericTrigger.CreateFallbackState(data, triggernum, state)
   local event = events[data.id][triggernum];
 
   Private.ActivateAuraEnvironment(data.id, "", state);
-  local firstTrigger = data.triggers[1].trigger
+  local trigger = data.triggers[triggernum].trigger
   if (event.nameFunc) then
-    local ok, name = xpcall(event.nameFunc, geterrorhandler(), firstTrigger);
+    local ok, name = xpcall(event.nameFunc, Private.GetErrorHandlerUid(data.uid, L["Name Function (fallback state"]), trigger);
```
Comment (Nightwarden24) [inline WeakAuras/GenericTrigger.lua:4057]: It seems to be a strange order of quote, square and round brackets. It should be `L["Name Function (fallback state)"]`

File: WeakAuras/GenericTrigger.lua [hunk 3]
```diff
@@ -4053,28 +4052,28 @@ function GenericTrigger.CreateFallbackState(data, triggernum, state)
...
     state.name = ok and name or nil;
   end
   if (event.iconFunc) then
-    local ok, icon = xpcall(event.iconFunc, geterrorhandler(), firstTrigger);
+    local ok, icon = xpcall(event.iconFunc, Private.GetErrorHandlerUid(data.uid, L["Icon Function (fallback state"]), trigger);
```
Comment (Nightwarden24) [inline WeakAuras/GenericTrigger.lua:4061]: the same here

File: WeakAuras/GenericTrigger.lua [hunk 4]
```diff
@@ -4053,28 +4052,28 @@ function GenericTrigger.CreateFallbackState(data, triggernum, state)
...
     state.icon = ok and icon or nil;
   end
 
   if (event.textureFunc ) then
-    local ok, texture = xpcall(event.textureFunc, geterrorhandler(), firstTrigger);
+    local ok, texture = xpcall(event.textureFunc, Private.GetErrorHandlerUid(data.uid, L["Texture Function (fallback state"]), trigger);
```
Comment (Nightwarden24) [inline WeakAuras/GenericTrigger.lua:4066]: the same here

File: WeakAuras/GenericTrigger.lua [hunk 5]
```diff
@@ -4053,28 +4052,28 @@ function GenericTrigger.CreateFallbackState(data, triggernum, state)
...
     state.texture = ok and texture or nil;
   end
 
   if (event.stacksFunc) then
-    local ok, stacks = xpcall(event.stacksFunc, geterrorhandler(), firstTrigger);
+    local ok, stacks = xpcall(event.stacksFunc, Private.GetErrorHandlerUid(data.uid, L["Stacks Function (fallback state"]), trigger);
```
Comment (Nightwarden24) [inline WeakAuras/GenericTrigger.lua:4071]: the same here

File: WeakAuras/GenericTrigger.lua [hunk 6]
```diff
@@ -4053,28 +4052,28 @@ function GenericTrigger.CreateFallbackState(data, triggernum, state)
...
     state.stacks = ok and stacks or nil;
   end
 
   if (event.durationFunc) then
-    local ok, arg1, arg2, arg3, inverse = xpcall(event.durationFunc, geterrorhandler(), firstTrigger);
+    local ok, arg1, arg2, arg3, inverse = xpcall(event.durationFunc, Private.GetErrorHandlerUid(data.uid, L["Duration Function (fallback state"]), trigger);
```
Comment (Nightwarden24) [inline WeakAuras/GenericTrigger.lua:4076]: the same here

File: WeakAuras/WeakAuras.lua [hunk 7]
```diff
@@ -78,6 +78,48 @@ function WeakAuras.InternalVersion()
...
+    local prefix = ""
+    local data
+    if currentErrorHandlerId then
+      data = WeakAuras.GetData(currentErrorHandlerId)
+    elseif currentErrorHandlerUid then
+      data = Private.GetDataByUID(currentErrorHandlerUid)
+    end
+    if data then
+      Private.AuraWarnings.UpdateWarning(data.uid, "LuaError", "error",
+        L["This aura has caused a Lua error."] .. "\n" .. L["Install the addons Bugsack and BugGraber for detailed error logs."])
```
Comment (mrbuds) [inline WeakAuras/WeakAuras.lua:95]:
```suggestion
      Private.AuraWarnings.UpdateWarning(data.uid, "LuaError", "error",
        L["This aura has caused a Lua error."] .. "\n" .. L["Install the addons BugSack and BugGrabber for detailed error logs."])
```

Comment (mrbuds) [reply WeakAuras/WeakAuras.lua:95]: `/console scriptErrors 1` is not *that* bad to recommend, i don't know state of blizz'error frame for all versions, if i remember right on DF alpha it's paged like bugsack

Comment (mrbuds) [reply WeakAuras/WeakAuras.lua:95]: What about turning the 5th arg of UpdateWarning `printOnConsole` on? It's made to not spam

Comment (InfusOnWoW) [reply WeakAuras/WeakAuras.lua]: Done that. It'll probably lead to lots of complaining, similar to the warnings about protected clones. But whatever.

Comment (InfusOnWoW) [reply WeakAuras/WeakAuras.lua]: I did?

**Review by mrbuds — APPROVED** [review]

**Pull request merged**